.github/workflows/build.yml

```diff
@@ -27,7 +27,7 @@ jobs:
 
   build:
     concurrency:
-      group: ${{ github.job }}-${{ strategy.job-index }}-${{ github.event_name == 'pull_request' && github.ref || github.sha }}
+      group: build-${{ strategy.job-index }}-${{ github.event_name == 'pull_request' && github.ref || github.sha }}
```

Comment:
It seems that  `github.job` is not available in the concurrency block, so it's replaced with the hardcoded name... Now it should work as expected...